### ISSUE TITLE
flutter_gen Package Import Gets Deleted 

### ISSUE THREAD

**Issue by supposedlysam-bb** [fluttercommunity/import_sorter#34]
**Describe the bug**
Without a comment on the same line the `import 'package:flutter_gen/gen_l10n/translations.dart';` gets removed when running `flutter pub run import_sorter:main`.

**To Reproduce**
You can just add the import to your main.dart file after creating a new project. It won't be able to build, but running `flutter pub run import_sorter:main` after saving the file still deletes the import. _Note: even after there are no build errors it still deletes it_ 

**Expected behavior**
The import should not be deleted.

**Meta Information:**
 - Dart Version: *2.10.2*
 - `import_sorter` Version: *4.2.2*
 - Is it a Flutter project? *Yes*
 - What version of Flutter are you using (if flutter project): *1.22.2*

**Additional Context**
The import is sorted correctly if you add a comment to the same line. For example:
`import 'package:flutter_gen/gen_l10n/translations.dart';` // gets removed

`import 'package:flutter_gen/gen_l10n/translations.dart'; // import_sorter:keep` // gets sorted correctly

**Comment by Matt-Gleich** [comment]
Hey @supposedlysam-bb! Sorry to hear about this issue. I'll take a look at it this week and keep you posted here.

**Comment by Matt-Gleich** [comment]
@supposedlysam-bb would mind sending me your pubspec.lock and pubspec.yaml?

**Comment by arsamme** [comment]
@Matt-Gleich  Any update on this?
It deletes this import for me:
import 'package:flutter_gen/gen_l10n/app_localizations.dart';

**Comment by Matt-Gleich** [comment]
I wasn't able to reproduce this on my end at the time. Would you mind sending me the whole file that this breaks for?

**Comment by arsamme** [comment]
@Matt-Gleich Whole project or the widget I used this package in?

**Comment by Matt-Gleich** [comment]
If you can send the project, that would be great. The single file that the import breaks work as well.

**Comment by arsamme** [comment]
[project.zip](https://github.com/fluttercommunity/import_sorter/files/6136119/project.zip)
@Matt-Gleich Here you are

**Comment by Matt-Gleich** [comment]
Thank you @arsamme! I'll look into this tomorrow and let you know the progress here :)

**Comment by arsamme** [comment]
@Matt-Gleich  I'll appreciate🌹

**Comment by arsamme** [comment]
@Matt-Gleich The issue is (here)[https://github.com/fluttercommunity/import_sorter/blob/9bb4beec11f289f78a587c7d323c39d5fedc49e0/lib/sort.dart#L66].
Because flutter_gen is not dependency itself.

**Comment by Matt-Gleich** [comment]
Hey, @arsamme. Sorry for not being able to get to this today. I have a lot going on with school right now and will try to get this as soon as possible. Sorry for the slight delay!

**Comment by arsamme** [comment]
@Matt-Gleich np, feel free. I can create pr and fix this if you want

**Comment by Matt-Gleich** [comment]
If you feel up to it, that would be great! Lemme know if you have any questions :)

**Comment by arsamme** [comment]
@Matt-Gleich Would you please check this pr? #50

**Comment by arsamme** [comment]
@Matt-Gleich Did you check this?

**Comment by smofe** [comment]
Any updates on this?

**Comment by Matt-Gleich** [comment]
Hey everyone! Sorry for the hold-up here. I just pushed out v4.4.3 of import_sorter which should resolve this issue.

**Comment by smofe** [comment]
Unfortunately, the error still exists in v4.4.3.

**Comment by Matt-Gleich** [comment]
hmm, strange. It worked for all my tests. Mind sending your project here? Sorry about this

**Comment by smofe** [comment]
I can reproduce this by creating the flutter example project and adding the import
`import 'package:flutter_gen/gen_l10n/translations.dart';`
to the main.dart

pubspec.yaml:
```dependencies:
  flutter:
    sdk: flutter
  flutter_localizations:
    sdk: flutter
  intl:


  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^1.0.2

dev_dependencies:
  flutter_test:
    sdk: flutter
  import_sorter: ^4.4.3```

**Comment by bartekpacia** [comment]
I also encounter this issue, because of it I had to stop using `import_sorter` in my project:(

**Comment by Matt-Gleich** [comment]
New fix out in v4.5.0. The sorting algorithm used to get package imports as long as the name was in the `dependencies` section of your pubspec.yaml. That is no longer the case so dev dependencies or other packages simply not listed will be put under package imports. Lemme know if this fixes it for you guys.

**Comment by bartekpacia** [comment]
Works flawlessly now! Thanks a lot :)
